base/compiler/abstractinterpretation.jl

```diff
@@ -327,6 +343,7 @@ function abstract_call_method(interp::AbstractInterpreter, method::Method, @nosp
                 # avoid widening when detecting self-recursion
                 # TODO: merge call cycle and return right away
                 if call_result_unused(sv)
+                    add_remark!(interp, sv, "Widening unused call result early to avoid unbounded recursion")
```

Comment:
```suggestion
                    add_remark!(interp, sv, "Widening unused call result early to avoid unbounded self-recursion")
```

Would be good if the annotations are unique so that one can grep for them. Not sure if my recommendation is the best here.

Reply:
As commented below, no widening is happening here (that's why we're okay with returning this early)

```suggestion
                    add_remark!(interp, sv, "Delaying call result analysis until after optimization due to bounded recursion")
```

Reply:
Are we really delaying analysis, aren't we just skipping it entirely? When will we come back to do this analysis?

Reply:
We're skipping this here because we know it's already being analyzed and we don't care about the result.

Reply:
Ok, but we never go back to actually annotate that result, right? In that case, I'd like to inform the user that the return type they'd be seeing in Cthulhu may not match what inference knows about this callsite. Maybe something like: "Unbounded recursion detected with unused result. Annotated return type may be wider than true result".

Reply:
This is bounded recursion (in the type domain), but yeah, we're putting 'Any' here specifically because we know it can't be used by the optimizer anyways, and thus want to avoid the re-work costs of putting the real type here later.

Reply:
So "Unbounded recursion detected with unused result. Annotated return type may be wider than true result."? Or just leave off the remark entirely and have Cthulhu check when it seems a situation like this?